### DISCUSSION TITLE
[Announcement] Setup & Homework 0

### DISCUSSION THREAD

**Discussion by goodtaeeun** [prosyslab-classroom/cs424-program-reasoning#297] [Announcements]
Hi, everyone.
I am Tae Eun, the TA of the course CS492 for this semester.

## Setup
Here are some things you should set up for this course.
- Submit your GitHub account: https://forms.gle/Ui6HrLWBivwwNTmLA
- Join the [Gradescope](https://www.gradescope.com/). We have sent the invitation mail based on the KLMS attendance book.
  - **We will send it again on Wednesday**. If you didn't get the invitation, please register and wait until the next Wednesday.
- Students will use OCaml and several tools for programming assignments. Here is the setup guide
  - [Guide for setup and OCaml programming](https://docs.google.com/presentation/d/1rS39M1dui3khqTc6joWniNrxH3x87nGgOrWHiMAL4DU)
  -  KCLOUD VM, one of your choices for the working environment, is not prepared yet but will be provided soon.

## HW0
Homework 0 includes **two programming exercises** and **one writing**.

Here are the GitHub classroom links for each assignment:

- Homework 0-1: https://classroom.github.com/a/bPNjCr1L
- Homework 0-2: https://classroom.github.com/a/TSWUiPAG
- Homework 0-3: https://classroom.github.com/a/k41fxszZ

Note that HW0 is not mandatory, but is intended to help you get started with the course smoothly.

### Homework 0-1 & 0-2

Homework 0-1 and 0-2 are programming exercises for those unfamiliar with OCaml and Git.

Feel free to skip these assignments if you are already used to both.

There are no deadlines because we won't grade them.

### Homework 0-3

Homework 0-3 is a writing assignment. 

Watch the video([KR](https://www.youtube.com/watch?v=oippSXvxUlw&feature=youtu.be), [EN](https://www.youtube.com/watch?v=HeQX2HjkcNo&t=2)) explaining undecidability, and write a critique about it. 

You should submit it at [Gradescope](https://www.gradescope.com/) in **PDF** format.

This assignment is due **9.8(Fri)**. As mentioned above, this is not graded. However, if you are aiming for a reward, be sure to submit it before the due date.

If you have any questions, feel free to ask the TA.

Thanks.

**Comment by goodtaeeun** [comment]
FYI, a typo is edited.
9/9 -> 9/8

**Comment by yeonhee-ryou** [comment]
'+ KCLOUD VM will be available next week (after course registration period)

**Comment by 08kmc09** [comment]
I didn’t receive any invitation email from it. Did I miss it? There was no e-mail with gradescope from Monday (8/28) ~ to Wednesday (8/30)

**Comment by GetRusty** [comment]
In the pdf file of homework 0-2, Problem 13, we are asked to write a function about an AVL Tree. But I don't see any AVL-related source or test code in my homework repository, nor in the dune file. Am I missing something?

**Comment by goodtaeeun** [comment]
> I didn’t receive any invitation email from it. Did I miss it? There was no e-mail with gradescope from Monday (8/28) ~ to Wednesday (8/30)

@08kmc09  I have sent the invitation email again. Please let me know if you have not received the email this time.

**Comment by goodtaeeun** [comment]
> In the pdf file of homework 0-2, Problem 13, we are asked to write a function about an AVL Tree. But I don't see any AVL-related source or test code in my homework repository, nor in the dune file. Am I missing something?

@GetRusty  You can ignore Problem 13. Sorry for the confusion.

**Comment by uk-jeong** [comment]
@goodtaeeun  I have not received the Gradescope invitation email yet. Should I contact separately?

**Comment by goodtaeeun** [comment]
@uk-jeong I am sorry. One student was accidentally omitted from the list, and it was you. I have sent you the invitation right now. Please feel free to inform me in case of further issues.
